res/shader/surface.inc.wgsl

```diff
@@ -40,6 +51,26 @@ fn modulo(a: i32, b: i32) -> i32 {
     return select(c, c+b, c < 0);
 }
 
+fn get_storage_index(ipos: vec2<i32>) -> StorageIndex {
+    let x = modulo(ipos.x, i32(u_Surface.texture_scale.x));
+    let y = modulo(ipos.y, i32(u_Surface.texture_scale.y));
+    // let x = ipos.x;
+    // let y = ipos.y;
+    let index = u32(y * i32(u_Surface.texture_scale.x) + x);
+
+    var si: StorageIndex;
+    si.index = index / 4u;
+    si.shift = 8u * (index % 4u);
+    si.mask = 0xFFu << si.shift;
+
+    return si;
+}
+
+fn get_storage_height(ipos: vec2<i32>) -> f32 {
```

Comment:
let's be consistent in naming things. The nearby `get_lod_height` returns the height normalized to `u_Surface.texture_scale.z`. This function should also do it.